root/etc/cont-init.d/98-julia

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/with-contenv bash
+
+gh_latest() {
+  curl -s https://api.github.com/repos/$1/$2/releases/latest \
+    | grep "tag_name" \
+    | cut -d : -f 2,3 \
+    | tr -d '\"\ v,'
+}
+
+# replace these with specific versions if needed (i.e 1.7.2 for Julia)
+JULIA_VERSION=$(gh_latest julialang julia)
+
+# Julia
+curl -so julia.tar.gz https://julialang-s3.julialang.org/bin/linux/x64/`echo ${JULIA_VERSION} | cut -d. -f 1,2`/julia-${JULIA_VERSION}-linux-x86_64.tar.gz
+tar fxz julia.tar.gz
+rm -R julia.tar.gz
+mv julia-${JULIA_VERSION}/bin/julia /usr/local/bin
```

Comment:
this should be done in a tmp folder and cleaned up after extract and move

Reply:
alternatively, it would be nice to have some versioning enabled via either:
1. Check for updates and only download the new tarball if there is an update, or
2. Ship the tarball in the image, push updates to mod image when there is an update upstream (preferred, we can help with setting the update trigger on jenkins)

Reply:
Which directory would be best? I'm not that well versed with docker, would just creating a `julia_dir` folder in `/tmp` work?
For now I've gone with `/home/root`. Is that a bad practice?
The julia directory needs to persist for the program to work, but the `tar.gz` is removed.

Reply:
I don't see where you're telling the app where its folder is located. I only see moving the bin into the local bin folder. `/home/root` is definitely not the right place as code server is not running as root, and the user's home folder is not set to it (and it's not persistent).

Reply:
I replaced the move command with a symlink. The new command is `ln -s $(pwd)/julia-${JULIA_VERSION}/bin/julia /usr/local/bin/julia`. So `/usr/local/bin/julia` should point to wherever we install it.
So now the workflow is download -> extract -> remove tar -> symlink to `/usr/local/bin/julia`.

If `/home/root` is not a good place, where else could we store it? As I said, I am quite new to Docker development. What is the standard way of keeping program files?

Reply:
I'm looking into julia install instructions and also see another issue, the binary you're pulling is x86_64 only, which is not going to work on arm32v7 and arm64v8 arches for these images